hazelcast/src/main/java/com/hazelcast/client/impl/ClientDelegatingFuture.java

```diff
@@ -54,14 +54,18 @@
 
     final boolean deserializeResponse;
     private final ClientMessageDecoder clientMessageDecoder;
-    private volatile Object decodedResponse = VOID;
+    private volatile Object decodedResponse;
 
     public ClientDelegatingFuture(ClientInvocationFuture clientInvocationFuture,
                                   SerializationService serializationService,
                                   ClientMessageDecoder clientMessageDecoder, V defaultValue, boolean deserializeResponse) {
-        super(serializationService, clientInvocationFuture, defaultValue);
+        super(serializationService, clientInvocationFuture, defaultValue, false);
```

Comment:
We want to ensure that `decodedResponse` is initialized to `VOID` before the dependent action in `whenComplete` is executed. If the delegate future (`clientInvocationFuture`) is completed concurrently, it may be the case that the dependent action is executed before the constructor is done -> if the dependent was already registered in super constructor it might observe `decodedResponse` as `null` instead of `VOID` and break the value resolution logic (client message decoding, deserialized value caching etc).

Reply:
Cool, thanks!